lib/request_started_on_middleware.rb

```diff
@@ -0,0 +1,57 @@
+# Add to config/application.rb:
+#
+#     config.middleware.use 'RequestStartedOnMiddleware'
+#
+class RequestStartedOnMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    start_request(env['PATH_INFO'], Time.now.utc)
+    @app.call(env)
+  ensure
+    complete_request
+  end
+
+  def start_request(path, started_on)
+    Thread.current[:current_request] = path
+    Thread.current[:current_request_started_on] = started_on
+  end
+
+  def complete_request
+    Thread.current[:current_request] = nil
+    Thread.current[:current_request_started_on] = nil
+  end
+
+  def self.long_running_requests
+    requests = []
+    timed_out_request_started_on = request_timeout.ago.utc
+
+    relevant_thread_list.each do |thread|
+      request    = thread[:current_request]
+      started_on = thread[:current_request_started_on]
+
+      # There's a race condition where the complete_request method runs in another
+      # thread after we set one or more of the above local variables. The fallout
+      # of this is we return a false positive for a request that finished very close
+      # to the 2 minute timeout.
+      if request.present? && started_on.kind_of?(Time) && timed_out_request_started_on > started_on
+        duration = (Time.now.utc - started_on).to_f
```

Comment:
I think we can "cache" `Time.now` in a local variable above with `request` and `started_on` instead of looking it up every iteration of the loop.

Reply:
Also something I noticed just now, is the `.utc` calls done through out these changes even necessary?  We never log the raw time, and only the difference between the start and current time when logging long requests.  I only mention this because we are trying to keep it as minimal/lightweight as possible, and calls to `.utc` seem pointless when dealing with time diffs.

Might be something that `rubocop` complained about though, so just something I noticed.

Reply:
👍 Yeah, since we're writing and reading the time, there's no need to store it as utc as long as it's always local